spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkAggregates.java

```diff
@@ -48,31 +48,20 @@ public static Expression convert(AggregateFunc aggregate) {
         case COUNT:
           Count countAgg = (Count) aggregate;
           if (countAgg.isDistinct()) {
-            // manifest file doesn't have count distinct so this can't be converted to push down
-            return null;
-          }
-
-          if (countAgg.column() instanceof NamedReference) {
-            return Expressions.count(SparkUtil.toColumnName((NamedReference) countAgg.column()));
-          } else {
             return null;
           }
+          assert (countAgg.column() instanceof NamedReference);
+          return Expressions.count(SparkUtil.toColumnName((NamedReference) countAgg.column()));
```

Comment:
I think this was actually better before. What was the rationale for converting to use `assert`? We don't use that elsewhere and I think the correct behavior is to return `null` when something is unexpected and can't be converted.

Reply:
Oh, I see that this was from the other `SparkAggregates`. I think we should use the `if`/`else` behavior.

Reply:
Changed to `if/else`